pallets/sponsorship/src/tests.rs

```diff
@@ -2519,3 +2519,303 @@ fn sponsor_call_is_invalid_if_sponsor_account_is_running_low() {
+			})),
+			Box::new(RuntimeCall::Uniques(pallet_uniques::Call::create {
+				collection: 2u32,
+				admin: user,
+			})),
+		];
+
+		let sponsor_for_uniques_create_call = Box::new(RuntimeCall::SponsorshipModule(Call::sponsor_for {
+			pot: 3,
+			calls: unique_create_call.clone(),
+		}));
+
+		let pre_dispatch_details = ChargeSponsor::<Test>::default()
+			.pre_dispatch(
+				&user,
+				&sponsor_for_uniques_create_call,
+				&sponsor_for_uniques_create_call.get_dispatch_info(),
+				0,
+			)
+			.ok();
+		assert!(SponsorshipModule::sponsor_for(RuntimeOrigin::signed(user), pot, unique_create_call.clone()).is_err());
+
+		assert_ok!(ChargeSponsor::<Test>::post_dispatch(
+			pre_dispatch_details,
+			&sponsor_for_uniques_create_call.get_dispatch_info(),
+			&().into(),
+			0,
+			&DispatchResult::Ok(())
+		));
+
+		let user_details_post_dispatch = User::<Test>::get(pot, user).unwrap();
+		let pot_details_post_dispatch = Pot::<Test>::get(pot).unwrap();
+
+		let fee = pot_details_post_dispatch.fee_quota.balance() - pot_details.fee_quota.balance();
+		assert_ne!(fee, 0);
+		assert_eq!(
+			user_details_post_dispatch.fee_quota.balance() - user_details.fee_quota.balance(),
+			fee
+		);
+
+		assert_eq!(user_details_post_dispatch.reserve_quota.balance(), 0);
+		assert_eq!(pot_details_post_dispatch.reserve_quota.balance(), 0);
+
+		System::assert_last_event(
+			Event::TransactionFeePaid {
+				sponsor: pot_details.sponsor,
+				fee,
+			}
+			.into(),
+		);
+		assert_eq!(
+			Balances::free_balance(pot_details.sponsor),
+			pot_reserve_quota
+				- fee - user_details_post_dispatch.reserve_quota.balance()
+				- PotDeposit::get()
+				- UserDeposit::get()
+		);
+		assert_eq!(
+			Balances::total_balance(&user_details.proxy),
+			user_details_post_dispatch.reserve_quota.balance()
+		);
+	});
+}
+
+#[test]
+fn compound_transaction_illegal_content_should_be_filtered() {
+	new_test_ext().execute_with(|| {
+		let pot = 3;
+		System::set_block_number(1);
+		let pot_fee_quota = 100_000_000_000;
+		let pot_reserve_quota = 100_000_000_000;
+		let pot_details = PotDetailsOf::<Test> {
+			sponsor: 1,
+			sponsorship_type: SponsorshipType::Uniques,
+			fee_quota: LimitedBalance::with_limit(pot_fee_quota),
+			reserve_quota: LimitedBalance::with_limit(pot_reserve_quota),
+			deposit: PotDeposit::get(),
+		};
+
+		Balances::make_free_balance_be(&pot_details.sponsor, pot_reserve_quota);
+
+		assert_ok!(SponsorshipModule::create_pot(
+			RuntimeOrigin::signed(pot_details.sponsor),
+			pot,
+			pot_details.sponsorship_type,
+			pot_details.fee_quota.limit(),
+			pot_details.reserve_quota.limit()
+		));
+
+		let user_fee_quota = pot_fee_quota / 10;
+		let user_reserve_quota = pot_reserve_quota / 10;
+
+		let user = 2u64;
+
+		assert_ok!(SponsorshipModule::register_users(
+			RuntimeOrigin::signed(pot_details.sponsor),
+			pot,
+			vec![user],
+			user_fee_quota,
+			user_reserve_quota
+		));
+		let _user_details = User::<Test>::get(pot, user).unwrap();
```

Comment:
You can remove this line just to make the test focused on its purpose. `register_users` is tested in our other unit tests and if it's ok we can proceed.